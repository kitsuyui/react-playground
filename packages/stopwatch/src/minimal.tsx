import React from 'react'

import { zeroPad2, zeroPad3 } from './utils'

import type { StopwatchProps } from './types'
export type * from './types'
export * from './container'

export const toLabel = (value: number) => {
  const minutes = Math.floor(value / 60)
  const seconds = value % 60 | 0
  const milliseconds = ((value % 1) * 1000) | 0
  return `${zeroPad2(minutes)}:${zeroPad2(seconds)}.${zeroPad3(milliseconds)}`
}

export const MinimalStopwatch: React.FC<StopwatchProps> = (
  props
): JSX.Element => {
  const { elapsedTime, running, toggle, reset } = props

  return (
    <div
      style={{
        width: '6em',
        fontFamily: 'monospace',
      }}
    >
      <form>
        <span>{toLabel(elapsedTime)}</span>
        <br />
        <button
          type="submit"
          name="reset"
          onClick={(e) => {
            e.preventDefault()
            reset()
          }}
        >
          Reset
        </button>
        <button
          type="submit"
          onClick={(e) => {
            e.preventDefault()
            toggle()
          }}
        >
          {running ? 'Stop' : 'Start'}
        </button>
      </form>
    </div>
  )
}
