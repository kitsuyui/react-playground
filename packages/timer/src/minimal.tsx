import React from 'react'

import type { TimerProps } from './types'
export type * from './types'
export * from './container'

export function toLabel(value: number) {
  const twoDigit = new Intl.NumberFormat('en-US', {
    minimumIntegerDigits: 2,
  })
  const threeDigit = new Intl.NumberFormat('en-US', {
    minimumIntegerDigits: 3,
  })

  const minutes = Math.floor(value / 60)
  const seconds = value % 60 | 0
  const milliseconds = (value % 1) * 1000
  return `${twoDigit.format(minutes)}:${twoDigit.format(
    seconds
  )}.${threeDigit.format(milliseconds)}`
}

export const MinimalTimer: React.FunctionComponent<TimerProps> = (
  props
): JSX.Element => {
  const { remaining, running, incrementTimerValue, toggle, reset } = props

  return (
    <div
      style={{
        border: '1px solid black',
        width: '6em',
      }}
    >
      <form>
        <span>{toLabel(remaining)}</span>
        <br />
        <button
          type="submit"
          name="incrementMinute"
          onClick={(e) => {
            e.preventDefault()
            incrementTimerValue(60)
          }}
        >
          +分
        </button>
        <button
          type="submit"
          name="incrementSecond"
          onClick={(e) => {
            e.preventDefault()
            incrementTimerValue(1)
          }}
        >
          +秒
        </button>
        <br />
        <button
          type="submit"
          name="reset"
          onClick={(e) => {
            e.preventDefault()
            reset()
          }}
        >
          Reset
        </button>
        <button
          type="submit"
          onClick={(e) => {
            e.preventDefault()
            toggle()
          }}
        >
          {running ? 'Stop' : 'Start'}
        </button>
      </form>
    </div>
  )
}
