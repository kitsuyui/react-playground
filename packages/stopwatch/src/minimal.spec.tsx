import { describe, expect, it, vi } from 'vitest'

import { render } from '@testing-library/react'

import { MinimalStopwatch } from './minimal'

describe('MinimalStopwatch', () => {
  it('renders correctly', () => {
    const props = {
      elapsedTime: 0,
      running: false,
      toggle: vi.fn(),
      reset: vi.fn(),
      start: vi.fn(),
      stop: vi.fn(),
    }
    const { asFragment } = render(<MinimalStopwatch {...props} />)
    expect(asFragment()).toMatchSnapshot()
  })
})
