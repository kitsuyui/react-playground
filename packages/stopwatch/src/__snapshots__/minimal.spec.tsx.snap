// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`MinimalStopwatch > renders correctly 1`] = `
<DocumentFragment>
  <span
    style="width: 6em; font-family: monospace;"
  >
    00:00.000
  </span>
  <button
    type="button"
  >
    Reset
  </button>
  <button
    type="button"
  >
    Start
  </button>
</DocumentFragment>
`;
